src/screens/Profile/Header/GrowableBanner.tsx

```diff
@@ -0,0 +1,149 @@
+  scrollY,
+  children,
+}: {
+  scrollY: SharedValue<number>
+  children: React.ReactNode
+}) {
+  const isFetching = useIsProfileFetching()
+  const delayedIsFetching = useDelayedValue(isFetching, 500)
+
+  const animatedStyle = useAnimatedStyle(() => ({
+    transform: [
+      {
+        scale: interpolate(scrollY.value, [-150, 0], [2, 1], {
+          extrapolateRight: Extrapolation.CLAMP,
+        }),
+      },
+    ],
+  }))
+
+  const animatedBlurViewProps = useAnimatedProps(() => {
+    return {
+      intensity: interpolate(
+        scrollY.value,
+        [-200, -15],
+        [80, 0],
+        Extrapolation.CLAMP,
+      ),
+    }
+  })
+
+  const animatedSpinnerStyle = useAnimatedStyle(() => {
+    return {
+      display: scrollY.value < 0 ? 'flex' : 'none',
+      opacity: interpolate(
+        scrollY.value,
+        [-60, -15],
+        [1, 0],
+        Extrapolation.CLAMP,
+      ),
+      transform: [
+        {translateY: interpolate(scrollY.value, [-150, 0], [-75, 0])},
+        {rotate: '90deg'},
+      ],
+    }
+  })
+
+  return (
+    <>
+      <Animated.View
+        style={[
+          a.absolute,
+          {left: 0, right: 0, bottom: 0},
+          {height: 150},
+          {transformOrigin: 'bottom'},
+          animatedStyle,
+        ]}>
+        {children}
+        <AnimatedBlurView
+          style={[a.absolute, a.inset_0]}
+          tint="dark"
+          animatedProps={animatedBlurViewProps}
+        />
+      </Animated.View>
+      <View style={[a.absolute, a.inset_0, a.justify_center, a.align_center]}>
+        <Animated.View style={[animatedSpinnerStyle]}>
+          <ActivityIndicator
+            key={delayedIsFetching ? 'spin' : 'stop'}
+            size="large"
+            color="white"
+            animating={delayedIsFetching}
+            hidesWhenStopped={false}
+          />
+        </Animated.View>
+      </View>
+    </>
+  )
+}
+
+function useIsProfileFetching() {
+  // are any of the profile-related queries fetching?
+  return [
+    useIsFetching({queryKey: [FEED_RQKEY_ROOT]}),
+    useIsFetching({queryKey: [FEEDGEN_RQKEY_ROOT]}),
+    useIsFetching({queryKey: [LIST_RQKEY_ROOT]}),
+    useIsFetching({queryKey: [STARTERPACK_RQKEY_ROOT]}),
+  ].some(isFetching => isFetching)
+}
+
+// stayed true for at least `delay` ms before returning to false
+function useDelayedValue(value: boolean, delay: number) {
+  const [prevValue, setPrevValue] = useState(value)
+  const [isDelayed, setIsDelayed] = useState(false)
+  const timeoutRef = React.useRef<ReturnType<typeof setTimeout>>()
+
+  useEffect(() => {
+    if (value !== prevValue) {
+      setPrevValue(value)
+      if (!value) {
+        setIsDelayed(true)
+        timeoutRef.current = setTimeout(() => setIsDelayed(false), delay)
+      }
+    }
+  }, [value, prevValue, delay])
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timeoutRef.current)
+    }
+  }, [])
+
+  return isDelayed ? true : prevValue
+}
```

Comment:
I really don't like this, is there a more elegant way to achieve this? I want the spinner to keep spinning for half a second while it animates back up

Reply:
might have a decent way of dealing with this, mentioned something in slack